### DISCUSSION TITLE
Strict-Transport-Security" HTTP header

### DISCUSSION THREAD

**Discussion by AndreScalaPT** [nextcloud/all-in-one#1697] [Questions] [answered]
Ho everyone,

For the last three weeks I've been trying to do what Nextcloud asks me: " The "Strict-Transport-Security" HTTP header is not set to at least "15552000" seconds. For enhanced security, it is recommended to enable HSTS as described in the security tips ↗."

I've read hundreds of questions about this and all the documentation, I've tried "/var/lib/docker/volumes/nextcloud_aio_apache" but no config file.

And force the redirect to HTTPS is in the same file?

Thank you,
kind regards,
André

**Comment by Zoey2936** [comment]
Are you using a reverse proxy?

**Reply by AndreScalaPT** [reply]
Hi, I’m using Cloudflare with a tunnel.

**Reply by Zoey2936** [reply]
you could try to enable this here (sorry that the screenshot is in German):
![grafik](https://user-images.githubusercontent.com/75573284/211168806-dcef679a-93e7-46a5-9d39-77a075dfc524.png)

**Reply by AndreScalaPT** [reply]
I never thought of this solution !!!!
So many hours creating/editing new installations and it was so simple!

Thank you very much,
Kind regards,
André

**Reply by stevenharris-dev** [reply]
Thank you so much! I have been trying to fix this issue, looking at my reverse proxy (caddy), docker configs, and everything. Of course, this issue would be found in the Cloudflare settings. I didn't understand why my Strict-Transport-Security kept being reset to 0.

**Reply by sabonim38** [reply]
I have tried to change it as above, but I still get the error...

The "Strict-Transport-Security" HTTP header is not set to at least "15552000" seconds. For enhanced security, it is recommended to enable HSTS as described in the [security tips ↗](https://docs.nextcloud.com/server/28/go.php?to=admin-security).

![image](https://github.com/nextcloud/all-in-one/assets/96001137/6047764d-e78d-48c5-94aa-1a9c989c7b4f)

Any other tips???

**Reply by kathool** [reply]
same I had it set for 8 days now and it still gives me the error
![image](https://github.com/user-attachments/assets/68aa937b-a194-4440-bf05-f78ce12f258e)
![image](https://github.com/user-attachments/assets/42a82efc-3a05-4365-936f-cdbfc45123da)

**Comment by szaimen** [comment, marked as answer]
Solution:

https://github.com/nextcloud/all-in-one/discussions/1697#discussioncomment-4621898

> you could try to enable this here (sorry that the screenshot is in German): ![grafik](https://user-images.githubusercontent.com/75573284/211168806-dcef679a-93e7-46a5-9d39-77a075dfc524.png)

**Comment by njotha** [comment]
for my nextcloud 30 installation (Ubuntu bullseye, Raspberry Pi 5 8GB) none of all those many suggestions to solve the HSTS Problem has helped -- until I had the absolutely crazy idea to add the famous line not only to the <VirtualHost *:443> section of my apache2 config file, but to the <VirtualHost *:80> section as well. Luckily, I don't care for the crazyness of ideas, I just look for their results...

**Comment by naelfe** [comment]
I am running NC 30.0.4 on docker (via unraid) with a nginx reverse proxy. I was wondering for quite some time why I got the warning, because my reverse proxy was configured correctly and I could also see the header in my session (firefox) to be set correctly.
Today I had some time and was stumbling upon this thread. 

While trying to check your different solutions, I also found this:
The ssl.conf in the nginx folder (user\appdata\nextcloud\nging\ssl.conf) had the following template in it.

```
# HSTS (ngx_http_headers_module is required) (63072000 seconds)
#add_header Strict-Transport-Security "max-age=63072000" always;
```

I simply removed the comment, restarted my docker and the warning is gone.
```
# HSTS (ngx_http_headers_module is required) (63072000 seconds)
add_header Strict-Transport-Security "max-age=63072000" always;
```

Maybe it will help somebody else as well :)

**Reply by alainq79** [reply]
Thank you verymuch. it's very helpfull.

**Reply by Quantumatum** [reply]
Thank you!!

**Comment by mateuszdrab** [comment]
Annoyingly, I have the same issue even though I have added the headers in the configuration of the ingress resource and I can verify them via curl.
`Strict-Transport-Security: max-age=15768000; includeSubDomains`

Despite the above, NC still shows the warning.
